lib/streamlit/runtime/app_session.py

```diff
@@ -383,12 +383,13 @@ def request_rerun(self, client_state: ClientState | None) -> None:
                 self._client_state.context_info.CopyFrom(client_state.context_info)
 
             rerun_data = RerunData(
-                client_state.query_string,
-                client_state.widget_states,
-                client_state.page_script_hash,
-                client_state.page_name,
+                query_string=client_state.query_string,
+                widget_states=client_state.widget_states,
+                page_script_hash=client_state.page_script_hash,
+                page_name=client_state.page_name,
                 fragment_id=fragment_id if fragment_id else None,
                 is_auto_rerun=client_state.is_auto_rerun,
+                cached_messages=list(client_state.cached_messages),
```

Comment:
It should probably be better to use `set` instead of `list` here since then we only check that specific hash in that cched_messages.This could give a slight performance improvement for situations when the number of cached messages is large.

Reply:
oh yeah, good point. I will change this

Reply:
Changed it to set 👍